sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala

```diff
@@ -30,13 +64,22 @@ class TableIdentifierParserSuite extends SparkFunSuite {
     // Illegal names.
     intercept[ParseException](parseTableIdentifier(""))
     intercept[ParseException](parseTableIdentifier("d.q.g"))
+  }
 
+  test("table identifier - keywords") {
     // SQL Keywords.
-    val keywords = Seq("select", "from", "where", "left", "right")
+    val keywords = Seq("select", "from", "where") ++ hiveNonReservedRegression
     keywords.foreach { keyword =>
       intercept[ParseException](parseTableIdentifier(keyword))
       assert(TableIdentifier(keyword) === parseTableIdentifier(s"`$keyword`"))
       assert(TableIdentifier(keyword, Option("db")) === parseTableIdentifier(s"db.`$keyword`"))
     }
   }
+
+  test("table identifier - non reserved keywords") {
+    // Hive keywords are allowed.
+    hiveNonReservedKeyword.foreach { nonReserved =>
+      assert(TableIdentifier(nonReserved) === parseTableIdentifier(nonReserved))
+    }
+  }
```

Comment:
Yeah, we need to fix this. The `HiveSqlParser` passes on queries it can't parse to Hive. I think we should disable this behavior after we have integrated the `SparkSqlParser` & `HiveSqlParser`. Could you create a JIRA ticket under https://issues.apache.org/jira/browse/SPARK-14776?